### ISSUE TITLE
Missing module dependencies or type declarations in this packages

### ISSUE THREAD

**Issue by FazioNico** [matrix-org/matrix-js-sdk#2310]
Hi everyone!
There is a missing module dependencies or type declarations in this packages. 
This does not allow compiling with typescript.

Have try to add with npm install but it return a 404
```
Error: node_modules/matrix-js-sdk/lib/crypto/CrossSigning.d.ts:5:27 - error TS2307: Cannot find module '@matrix-org/olm' or its corresponding type declarations.

5 import { PkSigning } from "@matrix-org/olm";
```

```
dev-fazio$ npm i @matrix-org/olm 
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@matrix-org%2folm - Not found
```

https://github.com/matrix-org/matrix-js-sdk/blob/6a51b02bedbc681d7da6192d03b9a87a70b24fd9/src/crypto/olmlib.ts#L26

**Comment by t3chguy** [comment]
I suggest reading https://github.com/matrix-org/matrix-js-sdk#end-to-end-encryption-support for how to install olm

**Comment by FazioNico** [comment]
> I suggest reading https://github.com/matrix-org/matrix-js-sdk#end-to-end-encryption-support for how to install olm

Thank you t3chguy for your answer.
The problem does not come when using the sdk but when compiling a Typescript project using the sdk.
Compilation is impossible because the `@matrix-org/olm` module or these type declarations are missing.

**Comment by t3chguy** [comment]
Indeed, not sure how a TS library is meant to have an optional featureset with an optional dependency without compilation errors

**Comment by FazioNico** [comment]
Yes I totaly agree.
I was able to test the sdk by modifying the sources of the packages with the optional definitions and the compilation works.
Here is the test I did. You will see, you can't serve or build project after npm install without modify sources files. 
There are a lot of type errors.

https://github.com/FazioNico/matrix-messaging

**Comment by CaLeBre05** [comment]
hi, I have the exact same error, bur I don't understand how it was fixed

**Comment by t3chguy** [comment]
You have to install Olm from https://gitlab.matrix.org/matrix-org/olm#javascript

**Comment by alexgaill** [comment]
Hi, did you find a way to avoid this error? When i try to install pre-buil olm, i have a new error:

```
An unexpected error occurred: "The property 'options.env['npm_config_____m_a_t_r_i_x___o_r_g___r_e_g_i_s_t_r_y_']' must be a string without null bytes. Received '\\x00h\\x00t\\x00t\\x00p\\x00s\\x00:\\x00/\\x00/\\x00g\\x00i
\\x00t\\x00l\\x00a\\x00b\\x00.\\x00m\\x00a\\x00t\\x00r\\x00i\\x00x\\x00.\\x00o\\x00r\\x00g\\x...".
```

**Comment by CaLeBre05** [comment]
Sorry, no. I gave up and will work on a fork of an already existing matrix client

**Comment by richvdh** [comment]
olm is no longer required.
